main.go

```diff
@@ -134,27 +140,27 @@ func (t *tagHandler) getTag(day int, hour int, currentTime time.Time) string {
 	windowTime := currentTime.Add(diffDays + diffHours)
 
 	if currentTime.After(windowTime) || currentTime.Equal(windowTime) {
-		return t.getCurrentTag(currentTime)
+		return t.getCurrentTag(currentTime), nil
 	}
 
-	return t.getPreviousTag(currentTime)
+	return t.getPreviousTag(currentTime), nil
 }
 
 func (t *tagHandler) getCurrentTag(currentTime time.Time) string {
-	return t.getImageDate(currentTime)
+	return floorToImageDay(currentTime).Format(tagFormat)
 }
 
-//getPreviousTag returns last week's tag according to the imageDay
+// getPreviousTag returns last week's tag according to the imageDay
 func (t *tagHandler) getPreviousTag(currentTime time.Time) string {
-	date := currentTime.AddDate(0, 0, -7)
-
-	return t.getImageDate(date)
+	if int(currentTime.Weekday()) < imageDay {
+		return t.getCurrentTag(currentTime)
+	}
```

Comment:
Might make sense to add a comment describing why this bit is needed. I would make a suggestion but I couldn't figure out a concise way to describe the reason.

Reply:
I'm also struggling to think of a concise explanation 😅